pkg/services/blockfetcher/blockfetcher.go

```diff
@@ -0,0 +1,353 @@
+package blockfetcher
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/url"
+
+	"github.com/nspcc-dev/neo-go/pkg/config"
+	"github.com/nspcc-dev/neo-go/pkg/core/block"
+	"github.com/nspcc-dev/neo-go/pkg/crypto/keys"
+	gio "github.com/nspcc-dev/neo-go/pkg/io"
+	"github.com/nspcc-dev/neo-go/pkg/network/bqueue"
+	"github.com/nspcc-dev/neo-go/pkg/network/payload"
+	"github.com/nspcc-dev/neo-go/pkg/services/oracle/neofs"
+	"github.com/nspcc-dev/neofs-sdk-go/client"
+	"github.com/nspcc-dev/neofs-sdk-go/object"
+	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
+	"go.uber.org/zap"
+)
+
+const (
+	ModeIndexSearch = "indexSearch"
+	ModeOidSearch   = "oidSearch"
+)
+
+// Ledger is an interface to Blockchain sufficient for Service.
+type Ledger interface {
+	AddBlock(block *block.Block) error
+	GetConfig() config.Blockchain
+	BlockHeight() uint32
+	AddHeaders(...*block.Header) error
+	HeaderHeight() uint32
+}
+
+// Service is a service that fetches blocks from NeoFS.
+type Service struct {
+	chain      Ledger
+	client     *client.Client
+	log        *zap.Logger
+	quit       chan bool
+	cfg        config.NeoFSBlockFetcher
+	queue      *bqueue.Queue
+	privateKey *keys.PrivateKey
+}
+
+// New creates a new BlockFetcherService.
+func New(chain Ledger, cfg config.NeoFSBlockFetcher, queue *bqueue.Queue, logger *zap.Logger) *Service {
+	privateKey, err := keys.NewPrivateKey()
+	if err != nil {
+		return nil
+	}
+	return &Service{
+		chain:      chain,
+		log:        logger,
+		quit:       make(chan bool),
+		cfg:        cfg,
+		queue:      queue,
+		privateKey: privateKey,
+	}
+}
+
+// Name implements the core.Service interface.
+func (bfs *Service) Name() string {
+	return "BlockFetcherService"
```

Comment:
It's not included into the list of network services, and thus BlockFetcher shouldn't implement this interface.